### PR TITLE
refactor: `platform` lib file

### LINKED ISSUE #5095: [Task]: Refactor `platform.ts` lib file

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

This code is old and should be touched up in accordance to our coding conventions.

## Changelog

```
- Remove platform lib file (move code to core app module)
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Closes #5095 

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [x] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

Test general app functionality, specifically things involving platform operations like importing / exporting Stronghold files.

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/app/interfaces/platform-event-map.interface.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { DashboardRoute } from '@core/router'
+
+import { IAppUpdateDownloadProgress } from './app-update-download-progress.interface'
+import { IAppVersionDetails } from './app-version-details.interface'
+
+export interface IPlatformEventMap {
+    'menu-logout': void
```
Comment (begonaalvarezd) [inline packages/shared/lib/core/app/interfaces/platform-event-map.interface.ts:7]: just as a comment, these are very desktop events, we might want to improve this in the future

**Review by begonaalvarezd — APPROVED** [review]
lgtm and works smooth on mobile👌🏼 plus the mobile crash is gone 💪🏼 
@MarkNerdi could you please review that everything also works as expected on desktop 🙏🏼 ?

File: packages/shared/lib/core/app/interfaces/platform-event-map.interface.ts (hunk 1 above)
Comment (maxwellmattryan) [reply packages/shared/lib/core/app/interfaces/platform-event-map.interface.ts:7]: Ah yes, this is true. Are some / any of these relevant for mobile?

Comment (begonaalvarezd) [reply packages/shared/lib/core/app/interfaces/platform-event-map.interface.ts:7]: Not that I can see, but we will have to check when we have the bindings working, just something to consider for the future

**Review by MarkNerdi — APPROVED** [review]
Works perfectly

**Pull request merged**